cmd/soroban-cli/src/invoke.rs

```diff
@@ -104,8 +104,8 @@ pub struct Cmd {
 
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    #[error("parsing argument {arg}: {error}")]
-    CannotParseArg { arg: String, error: StrValError },
+    #[error("parsing argument '{arg}' at index {index}: {error}")]
+    CannotParseArg { arg: String, error: StrValError, index: usize },
```

Comment:
I think this would be really cool to add. With the current implementation in this PR it doesn't print the index of the argument in the arg list to be given to the contract, it prints the index in the full argument list on the command line. This is somewhat confusing.

For example:
```
❯ soroban \
    invoke --id 1 \
    --wasm target/wasm32-unknown-unknown/release/soroban_hello_world_contract.wasm \
    --fn hello \
    --arg 'laksdjfklajsdlfkjasd'
error: parsing argument 'laksdjfklajsdlfkjasd' at index 8: parse error: value is not parseable to Symbol
```

Arguments:
- `0`: `invoke`
- `1`: `--id`
- `2`: `1`
- `3`: `--wasm`
- `4`: `target/wasm32-unknown-unknown/release/soroban_hello_world_contract.wasm`
- `5`: `--fn`
- `6`: `hello`
- `7`: `--arg`
- `8`: `'laksdjfklajsdlfkjasd'`

I think I would expect it to say argument 0 or 1, depending if the index is intended to start at zero or one.

Reply:
Not sure what is the best way to count the arg. Here is an attempt.